internal/impl/sql/input_sql_raw.go

```diff
@@ -0,0 +1,213 @@
+package sql
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"sync"
+
+	"github.com/benthosdev/benthos/v4/internal/shutdown"
+	"github.com/benthosdev/benthos/v4/public/bloblang"
+	"github.com/benthosdev/benthos/v4/public/service"
+)
+
+func sqlRawInputConfig() *service.ConfigSpec {
+	spec := service.NewConfigSpec().
+		// Stable(). TODO
+		Categories("Services").
+		Summary("Executes a select query and creates a message for each row received.").
+		Description(`Once the rows from the query are exhausted this input shuts down, allowing the pipeline to gracefully terminate (or the next input in a [sequence](/docs/components/inputs/sequence) to execute).`).
+		Field(driverField).
+		Field(dsnField).
+		Field(rawQueryField().
+			Example("SELECT * FROM footable WHERE user_id = $1;")).
+		Field(service.NewBloblangField("args_mapping").
+			Description("A [Bloblang mapping](/docs/guides/bloblang/about) which should evaluate to an array of values matching in size to the number of columns specified.").
+			Example("root = [ this.cat.meow, this.doc.woofs[0] ]").
+			Example(`root = [ meta("user.id") ]`).
+			Optional())
+	for _, f := range connFields() {
+		spec = spec.Field(f)
+	}
+
+	spec = spec.
+		Version("3.59.0").
+		Example("Consumes an SQL table using a query as an input.",
+			`
+Here we preform an aggregate over a list of names in a table that are less than 3600 seconds old.`,
+			`
+input:
+  sql_raw:
+    driver: postgres
+    dsn: postgres://foouser:foopass@localhost:5432/testdb?sslmode=disable
+    query: "SELECT name, count(*) FROM person WHERE last_updated < $1 GROUP BY name;"
+    args_mapping: |
+      root = [
+        now().ts_unix() - 3600
+      ]
+]
```

Comment:
Looks like this extra square bracket is causing the docs to fail linting (it's a check that verifies documentation examples are valid YAML).